src/util/chunk_fs.js

```diff
@@ -45,10 +45,9 @@ class ChunkFS extends stream.Transform {
 
     async _flush_buffers(callback) {
         if (this.q_buffers.length) {
-            const buffers_to_write = this.q_buffers;
+            await this.target_file.writev(this.fs_account_config, this.q_buffers);
             this.q_buffers = [];
             this.q_size = 0;
-            await this.target_file.writev(this.fs_account_config, buffers_to_write);
```

Comment:
Adding FSWorker _args_ref should already handle this case...

I think I prefer to keep resetting `this.q_buffers` **before** diving into any async code, because I worry that even if this might seem to work, it makes the code highly sensitive to any quirks in the nodejs stream/events behavior which can be a little hairy around the edges, even more than promises :) 

But we can still achieve the same and make super-duper sure that we keep ref on the buffers from this scope if you refer to it after writev:

```js
            const buffers_to_write = this.q_buffers;
            this.q_buffers = [];
            this.q_size = 0;
            await this.target_file.writev(this.fs_account_config, buffers_to_write);
            this._total_num_buffers += buffers_to_write.length;
```

Reply:
The purpose of the total_num_buffers is just a dummy to hold the ref in the Javascript scope?
I do not see any other use for this variable other than that, am I right?

Reply:
Yes